perspective_rectification/hough_transform_intersections.py

```diff
@@ -0,0 +1,102 @@
+import cv2
+import numpy as np
+from sklearn.cluster import KMeans
+
+from perspective_rectification.ploting import draw_lines
+from perspective_rectification.utils import convert_theta_to_two_points
+
+
+def hough_lines_main(image, contours):
+    """
+    Extract hough lines probabilistic, gather the cluster parameters, initialize the points of the clustering
+    fit the cluster, get the central points of the cluster and convert the x, y theta to 2d x y
+    """
+    height, width, _ = image.shape
+
+    # Prepare and extract the lines
+    min_line_length = 10
+    max_line_gap = 10
+    lines = cv2.HoughLinesP(contours, 1, np.pi / 180, 15, minLineLength=min_line_length, maxLineGap=max_line_gap)
+
+    clustering_parameters_modified, clustering_parameters = get_parameters_clustering(lines)
+
+    # Fit clustering
+    start_pts = gather_initial_points(image, clustering_parameters_modified)
+    kmeans = KMeans(n_clusters=4, init=start_pts, n_init=1)
+    kmeans.fit(clustering_parameters_modified)
+    averaged_lines = average_lines_clustered(clustering_parameters, kmeans.labels_)
+    draw_lines(image, averaged_lines, type_line="theta")
+
+    # Convert lines to 2d x y
+    out_lines = [convert_theta_to_two_points(x, y, theta) for x, y, theta in averaged_lines]
+    return out_lines
+
+
+def get_parameters_clustering(lines):
+    """
+    Merge x and y and extract theta from the slope of the difference of x times y
+    """
+    clustering_parameters_modified = []
+    clustering_parameters = []
+    for line in lines:
+        for x1, y1, x2, y2 in line:
+            if x2 == x1:
+                theta = np.sign(y2 - y1) * np.pi / 2
+            else:
+                p = (y2 - y1) / (x2 - x1)
+                theta = np.arctan(p)
+            x = (x1 + x2) * 0.5
+            y = (y1 + y2) * 0.5
+            # Instead of using x and y use an intersection of the 2d points line to the 0 x axis or y axis and
+            # instead of using Theta use the slope P
```

Comment:
Could you add this as a TODO instead? Currently it's not clear whether this is already implemented or not.